### PR TITLE
issue_5: add max encoder to KBinsDiscretizer

### PR DESCRIPTION
**Pull request by linjun9**
#### Reference Issues/PRs
issue reference: https://github.com/VishwaP98/lets-git-it-scikit-learn/issues/5
parent PR: https://github.com/VishwaP98/lets-git-it-scikit-learn/pull/1
#### What does this implement/fix? Explain your changes.
find the max value in each bin after KBinsDiscretizer allocates each entry to a specific bin.

### CODE REVIEW

**Comment by linjun9** [comment]
@VishwaP98 @RylandWang @shrutimistry @vishwaapatel @alalalau  review please 😄
Screen Shot 2021-03-06 at 9 26 39 PM

**Review by VishwaP98 — APPROVED** [review]

**Comment by VishwaP98** [comment]
looks good to me!

File: sklearn/preprocessing/tests/test_discretization.py [hunk 1]
```diff
@@ -239,6 +239,29 @@ def test_transform_mode():
     # check that original array is not mutated
     assert not np.array_equal(X , Xt)
 
+def test_transform_max():
+    expected_2bins = [[-1,   2.5, -3,  -0.5],
```
Comment (VishwaP98) [inline sklearn/preprocessing/tests/test_discretization.py:290]: can you initialize X here?

**Review by VishwaP98 — CHANGES_REQUESTED** [review]
initialize X inside of the test

File: sklearn/preprocessing/tests/test_discretization.py (hunk 1 above)
Comment (linjun9) [reply sklearn/preprocessing/tests/test_discretization.py:290]: no need for that. X is initialized at the top for this test_file

Comment (VishwaP98) [reply sklearn/preprocessing/tests/test_discretization.py:290]: Ignore this

**Review by VishwaP98 — APPROVED** [review]

**Review by shrutimistry — APPROVED** [review]

**Pull request merged**